### ISSUE TITLE
0.6.2 doesn't seem to produce all css

### ISSUE THREAD

**Issue by jwalton512** [tailwindcss/tailwindcss#518]
Using a basic tailwind setup:
```css
@import "tailwindcss/preflight";

/* Add custom base styles here... */

@import "tailwindcss/components";

/* Add custom components here... */

@import "tailwindcss/utilities";

/* Add custom utilities here... */
```
in 0.6.**2** the resulting file is 11kb and only appears to contain reset and container plugin
in 0.6.1 the resulting file is ~400kb and contains everything as normal

**Comment by adamwathan** [comment]
Yep noticed this on my own projects but can't replicate it in the tests 😩 I have a fix but don't understand *why* it fixes it, and I need to come up with a failing test to make sure I don't accidentally ever unfix it.

Will have something within a couple of hours!

**Comment by adamwathan** [comment]
Fixed in 0.6.3 but still need to understand the real cause of the problem in case there's a better fix:

https://github.com/tailwindcss/tailwindcss/commit/597bad6ee5a614547678a77342439985155c3711

**Comment by carlcassar** [comment]
@adamwathan Thank you for commenting on this so quickly. I sat down to use tailwind for the first time a couple of hours ago after getting excited about it thanks to Laracasts. Been scratching my head thinking I was doing something wrong 🤣

Still, excited to use it now. It looks brilliant. Keep up the good work.

**Comment by adamwathan** [comment]
Understand the cause and have pushed a fix to master that addresses it most accurately — the fix in 0.6.3 is fine and effectively the same.

For my own reference, the root cause is that PostCSS mutates container nodes when you try to move them somewhere else.

In this case I have some root tree `foo`, at-rule `bar` and a container of nodes `baz` that also happens to be of type `Root`. I want to move the nodes inside of `baz` to directly before the position of `bar` inside of `foo` like this:

```
bar.before(baz)
```

What I didn't realize is that this actually mutates `baz` — when this operation is complete the `baz` variable is now an empty root and its nodes have been moved.

This caused a problem specifically with Laravel Mix because for one reason or another it seems to run all of its PostCSS plugins twice, and on the second run the `baz` variable has been mutated because it's kept in memory as a closed over variable from a parent scope.

Cloning `baz` before moving it solved the problem, but a better solution is to make `baz` an array of nodes instead of a root in the first place, because then it doesn't get mutated.
